lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js

```diff
@@ -109,43 +101,26 @@ export default function BasicLTILaunchApp() {
   /**
    * Helper to handle thrown errors from from API requests.
    *
-   * @param {Error} e - Error object from request.
+   * @param {Error} error - Error object from request.
    * @param {ErrorState} state
    * @param {boolean} [retry=true] - Can the request be retried?
    * @param {string} [authUrl] -
    *   Authorization URL association with the API request
    */
-  const handleError = (e, state, retry = true, authUrl) => {
+  const handleError = (error, state, retry = true, authUrl) => {
     // Here we always set the authorization URL, but we could improve UX by
     // not setting it if the problem is not related to authorization (eg.
     // a network fetch error).
     setAuthUrl(authUrl || null);
 
-    if (
-      e instanceof APIError &&
-      e.errorCode &&
-      [
-        'blackboard_file_not_found_in_course',
-        'canvas_api_permission_error',
-        'canvas_file_not_found_in_course',
-        'canvas_group_set_not_found',
-        'canvas_group_set_empty',
-        'canvas_student_not_in_group',
-      ].includes(e.errorCode)
-    ) {
-      // In this case, we're dealing with an APIError from an API request to
-      // our own backend, of a known error code.
-      setError(e);
-      setErrorState(/** @type {ErrorState} */ (e.errorCode));
-    } else if (e instanceof APIError && !e.serverMessage && retry) {
-      // This is a special case expected by the back end. We're handling an
-      // APIError resulting from an API request, but there are no further
-      // details in the response body to guide us. This implicitly means that
-      // we're facing an authorization-related error.
+    if (isLTILaunchServerError(error)) {
+      setError(error);
+      setErrorState(/** @type {ErrorState} */ (error.errorCode));
+    } else if (isAuthorizationError(error) && retry) {
```

Comment:
I'm not entirely clear on the purpose of the `retry` part of this conditional check, as I believe `reply` is always true in cases in which we might encounter an authorization error here. I don't want to poke at too many things at once, however.